tasks/main.yml

```diff
@@ -1,26 +1,58 @@
 ---
-- name: Install openvpn
+- name: Install OpenVPN
   apt: 
     name: openvpn
 
-- name: Slurp openvpn config
+- name: Ensure base directory exists
+  file:
+    path: "{{ openvpn_client_base_dir }}"
+    state: directory
+
+- name: Slurp recommended client config
   slurp:
     src: "{{ hostvars[openvpn_client_server].openvpn_base_dir }}/{{ openvpn_client_name }}-{{ openvpn_client_server }}.ovpn"
   register: openvpn_config_file
   delegate_to: "{{ openvpn_client_server }}"
-  when: openvpn_client_server is defined
+  when: openvpn_client_local_config_path is not defined
 
-- name: Copy OpenVPN config to host 
+- name: Copy client config to host
   copy:
     content: "{{ openvpn_config_file['content'] | b64decode }}"
-    dest: "/etc/openvpn/{{ openvpn_client_name }}-{{ openvpn_client_server }}.conf"
-  when: openvpn_client_server is defined
-  notify: Restart openvpn service
+    dest: "{{ openvpn_client_config_path }}"
+  register: openvpn_client_config_stat
+  when: openvpn_client_local_config_path is not defined
+  notify: Restart OpenVPN service
 
-- name: Copy OpenVPN config to host 
+- name: Copy client config to host
   copy:
-    src: "files/{{ inventory_hostname }}/{{ openvpn_client_config_file_name }}"
-    dest: "/etc/openvpn/{{ openvpn_client_name }}.conf"
-  when: openvpn_client_server is not defined
-  notify: Restart openvpn service
+    src: "files/{{ inventory_hostname }}/{{ openvpn_client_local_config_path }}"
+    dest: "{{ openvpn_client_config_path }}"
+  register: openvpn_client_config_stat
+  when: openvpn_client_local_config_path is defined
+  notify: Restart OpenVPN service
+
+- name: Instantiate up-script
+  template:
+    src: up.sh.j2
+    dest: "{{ openvpn_client_base_dir }}/up.sh"
+    mode: 0755
+  notify: Restart OpenVPN service
+
+- name: Allow up-script to execute commands
+  lineinfile:
+    path: "{{ openvpn_client_config_path }}"
+    line: script-security 2
+  notify: Restart OpenVPN service
+
+- name: Add up-script to config
```

Comment:
You might be able to reduce the number of tasks by appending the two lines to the `openvpn_client_extra_config` list.

Reply:
Then this would also need to be done when overwriting `openvpn_client_extra_config`.

Reply:
Yes, I would just replace this line https://github.com/stuvusIT/openvpn-client/pull/2/files/cba0a6efd961249fe4df41dc146cc2be789d911a#diff-2444ad0870f91f17ca6c2a5e96b26823R57
with
```
with_items: "{{ openvpn_client_extra_config | combine(['script-security 2', 'up {{ openvpn_client_base_dir }}/up.sh']) }}"
```
One could also use `set_fact` or the general `vars` to create a combined version of the options.

Reply:
Note: `combine` is for dictionaries.

I added an according commit.